### PR TITLE
upgrade to 3.11

### PR DESCRIPTION
**Pull request by nathangoulding**
Upgrading our installer from Alpine 3.7 to Alpine 3.11, and testing the automatic rebuild of Alpine's initramfs, kernel, and modules if those bits change.

### CODE REVIEW

**Comment by nathangoulding** [comment]
Tests are green, and we're ready for initial code review. A few things before merging:

- We need to understand what we're doing with the Eclypsium module. I have an email thread going with the team. I'd prefer to fetch and add this in a more standard way, or remove it from osie and turn this into a workflow action. Thoughts here are welcome, but in all cases we'll need to address this.

- We will need to test this across our fleet to check backwards compatibility with all our systems. Any suggestions on how we can do this?

**Comment by mmlb** [comment]
>     * We need to understand what we're doing with the Eclypsium module. I have an email thread going with the team. I'd prefer to fetch and add this in a more standard way, or remove it from osie and turn this into a workflow action. Thoughts here are welcome, but in all cases we'll need to address this.

Workflow makes a lot of sense to me.

>     * We will need to test this across our fleet to check backwards compatibility with all our systems. Any suggestions on how we can do this?

Well since you are in a deep-dive groove want to resurrect the provision based tests? It's a bit hard to follow but can be simplified a lot now that api has batch endpoint, it cleans up a lot of the logic. We could also test preinstalls this way too.

File: installer/runner.sh [hunk 1]
```diff
@@ -81,9 +81,17 @@ phone_home_url=$(jq -r .phone_home_url "$metadata")
...
-	sleep 3
+for i in $(seq 5); do
+	echo "Waiting for docker to respond (${i})..." &&
+		docker info &&
+		break
+	sleep 5
 done
+if ! docker info; then
+	cat /var/log/docker.log
+	echo "Docker not responding, dropping to a shell..."
```
Comment (rgl) [inline installer/runner.sh:92]: is this installer/runner.sh file manually executed by a human?

is it automatically executed by a machine? if so, this should not drop into a shell.

File: installer/alpine/build.sh [hunk 2]
```diff
@@ -31,15 +33,15 @@ mv /etc/mkinitfs/features.d/virtio.modules.tmp /etc/mkinitfs/features.d/virtio.m
 
 # Make initramfs with features we think are spiffy
 # shellcheck disable=SC2016
-echo 'features="base ext2 ext3 ext4 keymap network packetrepo squashfs virtio eclypsium"' >/etc/mkinitfs/mkinitfs.conf
+echo 'features="base ext2 ext3 ext4 keymap network packetrepo squashfs virtio"' >/etc/mkinitfs/mkinitfs.conf
 kver=$(basename /lib/modules/*)
 mkinitfs -l "$kver"
-mkinitfs -o /assets/initramfs-vanilla "$kver"
+mkinitfs -o /assets/initramfs-"$arch" "$kver"
```
Comment (rgl) [inline installer/alpine/build.sh:39]: this is an odd way to quote a variable. isn't it nicer to the reader to use `"/assets/initramfs-$arch"` instead?

File: installer/alpine/init-x86_64 [hunk 3]
```diff
@@ -542,7 +644,7 @@ fi
 if [ -f "$ovl" ]; then
 	ebegin "Loading user settings from $ovl"
 	# create apk db and needed /dev/null and /tmp first
-	apk add --root $sysroot --initdb --quiet --no-progress
+	apk add --root $sysroot --initdb --quiet
```
Comment (rgl) [inline installer/alpine/init-x86_64:647]: why remove `--no-progress`? will apk to the right thing of not showing the progress when run in CI?

**Review by rgl — COMMENTED** [review]
Bump this PR to Alpine 3.12 too?

**Comment by mmlb** [comment]
> Bump this PR to Alpine 3.12 too?

maybe, one of the last bits of work here to verification on a bunch of our machines. Will see about updating first to and be able to knock both things out.

File: installer/runner.sh (hunk 1 above)
Comment (mmlb) [reply installer/runner.sh:92]: Runs automatically, drops to a shell when an issue happens and support and/or devs are able to poke around and debug.

File: installer/alpine/build.sh (hunk 2 above)
Comment (mmlb) [reply installer/alpine/build.sh:39]: yes

File: installer/runner.sh (hunk 1 above)
Comment (rgl) [reply installer/runner.sh:92]: Isn't this executed in a CI environment? Without user access.

**Comment by mrmrcoleman** [comment]
Hey @mmlb any update on this?

**Comment by mmlb** [comment]
closing, since we'll be going with 3.12 and have further work to do.
